polaris-react/src/components/Button/Button.tsx

```diff
@@ -200,6 +200,25 @@ export function Button({
     setDisclosureActive((disclosureActive) => !disclosureActive);
   }, []);
 
+  const handleClick = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      if (disabled) {
+        event.preventDefault();
+        return;
+      }
+      toggleDisclosureActive();
+    },
+    [disabled, toggleDisclosureActive],
+  );
+  const handleKeyDown = useCallback(
+    (event: React.KeyboardEvent<HTMLButtonElement>) => {
+      if (['Enter', ' '].includes(event.key) && disabled) {
+        event.preventDefault();
+      }
+    },
+    [disabled],
+  );
```

Comment:
I don't know off the top of my head, but I assumed using `aria-disabled` would have given us this functionality for free. This logic could probably be repeated for all Polaris elements that are disabled.